pkg/scheduler/core/generic_scheduler.go

```diff
@@ -389,12 +394,21 @@ func (g *genericScheduler) getLowerPriorityNominatedPods(pod *v1.Pod, nodeName s
 
 // numFeasibleNodesToFind returns the number of feasible nodes that once found, the scheduler stops
 // its search for more feasible nodes.
-func (g *genericScheduler) numFeasibleNodesToFind(numAllNodes int32) int32 {
-	if numAllNodes < minFeasibleNodesToFind || g.percentageOfNodesToScore <= 0 ||
-		g.percentageOfNodesToScore >= 100 {
+func (g *genericScheduler) numFeasibleNodesToFind(numAllNodes int32) (numNodes int32) {
+	if numAllNodes < minFeasibleNodesToFind || g.percentageOfNodesToScore >= 100 {
 		return numAllNodes
 	}
-	numNodes := numAllNodes * g.percentageOfNodesToScore / 100
+
+	if g.percentageOfNodesToScore <= 0 {
```

Comment:
Our [defaulting mechanism](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/apis/config/v1alpha1/defaults.go#L84), currently sets `PercentageOfNodesToScore` to 50 if it is not specified. You should remove that to let the scheduler adjust it dynamically.